runelite-client/src/main/java/net/runelite/client/plugins/twitch/TwitchPlugin.java

```diff
@@ -154,6 +187,228 @@ public void onConfigChanged(ConfigChanged configChanged)
+				//message.revalidate();
+			}
+
+		}
+
+	}
+
+	/**
+	 * @author JBerben
+	 *
+	 * Constructs a formatted message displaying the senders
+	 * color, name and all appropriate badges. Hopefully in the
+	 * future, we can add "TWITCH" as a ChatMessageType enum into
+	 * RuneLite's api. I'm not sure if it's allowed or not.
+	 *
+	 * Examples of params
+	 * 	badgeString: swagtwitchuser:f70000:subscriber/1,moderator/1
+	 *				 TWITCH-USER	COLOR	BADGE/RANK,BADGE2/RANK
+	 *
+	 * @param badgeString a string containing all the information
+	 *                    required in order to display the sender's
+	 *                    user handle, color and relevant Twitch badges.
+	 *
+	 * @return returns a formatted string that will be ready to send.
+	 * 			Examples:
+	 *
+	 * 	This is one clan chat message. First is cc motd, rank, senderName.
+	 * 	Second is the actual content of the message.
+	 * 	1st: OriginalX: 3, OriginalWidth: 486
+	 * 	2nd: OriginalX: 133, OriginalWidth: Dependant on message length.
+	 *	[<col=0000ff>Mining <img=20></col>] AustinDies:
+	 *	<col=7f0000>Or uhh 6, that is</col>
+	 *
+	 * 	This is what a fully constructed user badge handle would look like:
+	 *	Broadcaster>Mod>Sub>Turbo>Prime>BitCheer(Not currently supported) | Hierarchy
+	 *	<col=SENDERCOLOR><img=7><img=25>SENDERNAME</col>
+	 *	COLOR_OF_SENDER | SUB6 | PRIME | USER_NAME
+	 */
+	static String constructBadgeIcons(String badgeString)
+	{
+		StringBuilder sb = new StringBuilder();
+
+		//DEBUG
+		System.out.println(badgeString);
+
+		String senderName = badgeString.split(":")[0];
+		String senderColor = badgeString.split(":")[1];
+		String badges = badgeString.split(":")[2];
+
+		int senderSubRank = TwitchBadges.getCodeGivenRank("subscriber", getSubRank(badges)).getImgCode();
+
+		/** Create the badge string **/
+		// TODO: Enable bit cheer badges
+		sb.append("<col=" + senderColor + ">");
+		if (isBroadcaster(badges))
+		{
+			sb.append("<img=" + TwitchBadges.getCodeGivenRank("broadcaster", 1).getImgCode() + ">");
+		}
+		if (isModerator(badges))
+		{
+			sb.append("<img=" + TwitchBadges.getCodeGivenRank("moderator", 1).getImgCode() + ">");
+		}
+		if (senderSubRank != 0)
+		{
+			sb.append("<img=" + senderSubRank + ">");
+		}
+		if (isTurbo(badges))
+		{
+			sb.append("<img=" + TwitchBadges.getCodeGivenRank("turbo", 1).getImgCode() + ">");
+		}
+		if (isPrime(badges))
+		{
+			sb.append("<img=" + TwitchBadges.getCodeGivenRank("prime", 1).getImgCode() + ">");
+		}
+		sb.append(senderName + "</col>");
+		System.out.println("Constructed Badge Message: " + sb.toString());
+		return sb.toString();
+	}
+
+	static int getSubRank(String badgeString)
+	{
+		if (isSubscriber(badgeString))
+		{
+			for (String token : badgeString.split(","))
+			{
+				if (token.contains("subscriber"))
+				{
+					return Integer.valueOf(token.split("/")[1]);
+				}
+			}
+		}
+		// Not a subscriber.
+		return 0;
+	}
+
+	/** boolean functions to assist code readability **/
```

Comment:
I think all of these can be `private`? Update that if so; it's not likely they'd be used in other classes.